pxf/pxf-json/src/main/java/org/apache/hawq/pxf/plugins/json/JsonResolver.java

```diff
@@ -0,0 +1,259 @@
+import java.util.List;
+
+import org.apache.hawq.pxf.api.OneField;
+import org.apache.hawq.pxf.api.OneRow;
+import org.apache.hawq.pxf.api.ReadResolver;
+import org.apache.hawq.pxf.api.io.DataType;
+import org.apache.hawq.pxf.api.utilities.ColumnDescriptor;
+import org.apache.hawq.pxf.api.utilities.InputData;
+import org.apache.hawq.pxf.api.utilities.Plugin;
+import org.codehaus.jackson.JsonNode;
+
+/**
+ * This JSON resolver for PXF will decode a given object from the {@link JsonAccessor} into a row for HAWQ. It will
+ * decode this data into a JsonNode and walk the tree for each column. It supports normal value mapping via projections
+ * and JSON array indexing.
+ */
+public class JsonResolver extends Plugin implements ReadResolver {
+
+	private ArrayList<OneField> list = new ArrayList<OneField>();
+
+	public JsonResolver(InputData inputData) throws Exception {
+		super(inputData);
+	}
+
+	@Override
+	public List<OneField> getFields(OneRow row) throws Exception {
+		list.clear();
+
+		// key is a Text object
+		JsonNode root = JsonInputFormat.decodeLineToJsonNode(row.getKey().toString());
+
+		// if we weren't given a null object
+		if (root != null) {
+			// Iterate through the column definition and fetch our JSON data
+			for (int i = 0; i < inputData.getColumns(); ++i) {
+
+				// Get the current column description
+				ColumnDescriptor cd = inputData.getColumn(i);
+				DataType columnType = DataType.get(cd.columnTypeCode());
+
+				// Get the JSON projections from the column name
+				// For example, "user.name" turns into ["user","name"]
+				String[] projs = cd.columnName().split("\\.");
+
+				// Move down the JSON path to the final name
+				JsonNode node = getPriorJsonNode(root, projs);
+
+				// If this column is an array index, ex. "tweet.hashtags[0]"
+				if (isArrayIndex(projs)) {
+
+					// Get the node name and index
+					String nodeName = getArrayName(projs);
+					int arrayIndex = getArrayIndex(projs);
+
+					// Move to the array node
+					node = node.get(nodeName);
+
+					// If this node is null or missing, add a null value here
+					if (node == null || node.isMissingNode()) {
+						addNullField(columnType);
+					} else if (node.isArray()) {
+						// If the JSON node is an array, then add it to our list
+						addFieldFromJsonArray(columnType, node, arrayIndex);
+					} else {
+						throw new InvalidParameterException(nodeName + " is not an array node");
+					}
+				} else {
+					// This column is not an array type
+					// Move to the final node
+					node = node.get(projs[projs.length - 1]);
+
+					// If this node is null or missing, add a null value here
+					if (node == null || node.isMissingNode()) {
+						addNullField(columnType);
+					} else {
+						// Else, add the value to the record
+						addFieldFromJsonNode(columnType, node);
+					}
+				}
+			}
+		}
+
+		return list;
+	}
+
+	/**
+	 * Iterates down the root node to the prior JSON node. This node is used
+	 * 
+	 * @param root
```

Comment:
please fill up all the parameters in the javadoc. It causes the build to fail when compiling with java8.